loki/transformations/inline.py

```diff
@@ -597,4 +604,27 @@ def inline_marked_subroutines(routine, allowed_aliases=None, remove_imports=True
                 )
                 # Remove import if no further symbols used, otherwise clone with new symbols
                 import_map[impt] = impt.clone(symbols=new_symbols) if new_symbols else None
+
+        # Now move any callee imports we might need over to the caller
+        new_imports = set()
+        imported_module_map = {im.module: im for im in routine.imports}
```

Comment:
Are we case-safe here? Or should this be a `CaseInsensitiveDict` maybe, because we're subsequently comparing imports from the callee against imports from the caller, where module names might have different casing.

Reply:
Very good catch! Yes, this needs fixing, will push a fix on top of rebase.